src/ts/i18n.ts

```diff
@@ -0,0 +1,106 @@
+import { en, de } from './translations';
+import { Translation } from './translations/translation-types';
+
+
+// TODO: This import style should be removed after build tool upgrade
+// This hack gives us minimal build size
+import _merge = require('lodash/merge');
+import _cloneDeep = require('lodash/cloneDeep');
+
+
+// This gives saves us build size but will not work with functions
+// Better solution is requiered
+// let _cloneDeep = function(o: any): any {
+//   return JSON.parse(JSON.stringify(o))
+// };
+
+
+
+/**
+  This module will provide other modules with localized UI strings
+  via a singleton pattern.
+
+  Singleton is a root of all evil, but in this case we provide it as a sane
+  default. There is an escape hatch (`I18n` class) provided.
+
+  Assumptions:
+    - We will use english as a default language.
+    - User can supply another language via UI-config and we will respect that
+    - It is possible to translate UI internally,
+      and provide locale via a simple config string
+
+  Future work:
+    - It is possible to make use of getters / setters:
+      - will lead to better IDE support
+      - would be possible to utilize typescript built-in
+        interface and enum support. This will also reduce
+        a bundle size
+*/
+
+const predifinedLanguages: any = {
```

Comment:
Typo: `predefinedLanguages`

Reply:
Please don't use `any` types... better to omit it and let TS infer the type, e.g. here it will probably be `{ [name: string]: Translation }`... but ideally the type is correctly defined